### ISSUE TITLE
Crash in Split iOS SDK

### ISSUE THREAD

**Issue by matcartmill** [splitio/ios-client#283]
Hello,

We are experiencing a very rare crash related to the Split IO iOS SDK. Here is the stack trace:

```
Crashed: split-evt-mngr-process
0  Polaris                        0x3e1db4 closure #1 in DefaultSplitEventsManager.eventAlreadyTriggered(event:) + 4311621044 (<compiler-generated>:4311621044)
1  Polaris                        0x3a1760 thunk for @callee_guaranteed () -> () + 4311357280 (<compiler-generated>:4311357280)
2  Polaris                        0x3a12d0 thunk for @escaping @callee_guaranteed () -> () + 4311356112 (<compiler-generated>:4311356112)
3  libdispatch.dylib              0x481c _dispatch_client_callout + 20
4  libdispatch.dylib              0x13f8c _dispatch_sync_invoke_and_complete + 60
5  Polaris                        0x3e1c90 DefaultSplitEventsManager.eventAlreadyTriggered(event:) + 86 (SplitEventsManager.swift:86)
6  Polaris                        0x3ec8bc RetryableMySegmentsSyncWorker.fetchFromRemote() + 130 (RetryableSyncWorker.swift:130)
7  Polaris                        0x3ec004 closure #1 in BaseRetryableSyncWorker.start() + 59 (RetryableSyncWorker.swift:59)
8  Polaris                        0x3a12d0 thunk for @escaping @callee_guaranteed () -> () + 4311356112 (<compiler-generated>:4311356112)
9  libdispatch.dylib              0x2a84 _dispatch_call_block_and_release + 32
10 libdispatch.dylib              0x481c _dispatch_client_callout + 20
11 libdispatch.dylib              0x7968 _dispatch_queue_override_invoke + 756
12 libdispatch.dylib              0x15fe0 _dispatch_root_queue_drain + 388
13 libdispatch.dylib              0x167d8 _dispatch_worker_thread2 + 112
14 libsystem_pthread.dylib        0x3768 _pthread_wqthread + 216
15 libsystem_pthread.dylib        0xa74c start_wqthread + 8
```

We are running version `2.11.3`. We will try updating to the latest production-ready release which appears to be `2.11.4`, but I thought I would bring this to your attention anyhow, as the frequency at which this crash occurs is incredibly low, so I might not know for quite some time if upgrading to `2.11.4` resolved the issue.

Our current implementation for how Split it created is:

```swift
        client?.destroy()

        let config = SplitClientConfig()
        config.sdkReadyTimeOut = self.config.readyTimeOut // 5 seconds
        config.featuresRefreshRate = self.config.refreshInterval // 1 hour

        let matchingKey = clientKey()

        let key = Key(matchingKey: matchingKey ?? "default")
        let builder = DefaultSplitFactoryBuilder()
        let factory = builder
            .setApiKey(apiKey)
            .setKey(key)
            .setConfig(config)
            .build()

        self.client = factory?.client

        client?.on(event: .sdkReady) { [weak self] in
            guard let self = self else { return }
            context.execute { complete() }
            self.subscribers.notify(with: self)
        }

        client?.on(event: .sdkReadyTimedOut) { [weak self] in
            guard let self = self else { return }

            self.client?.destroy()
            self.client = nil

            if retryCount < self.config.factoryRetryCount {
                self.makeClient(deliverOn: context, retryCount: retryCount + 1, complete: complete)
            } else {
                context.execute { complete() }
            }
        }
```

I can provide more information if needed.

Thanks.

**Comment by agustinona** [comment]
Hi @matcartmill,

Thank you. A fix for this issue is included in version 2.11.4.

Please do let us know if you see an occurrence after the upgrade.

**Comment by matcartmill** [comment]
Thank you for confirming.
